src/runtime_src/tools/xclbinutil/SectionEmbeddedMetadata.cxx

```diff
@@ -30,5 +32,29 @@ SectionEmbeddedMetadata::~SectionEmbeddedMetadata() {
   // Empty
 }
 
+void
+SectionEmbeddedMetadata::marshalToJSON(char* _pDataSection,
+                                       unsigned int _sectionSize,
+                                       boost::property_tree::ptree& _ptree) const {
+  XUtil::TRACE("");
+  XUtil::TRACE("Extracting: EMBEDDED_METADATA");
+  XUtil::TRACE_BUF("Section Buffer", reinterpret_cast<const char*>(_pDataSection), _sectionSize);
+
+  // Determine if there is something to do, if not then leave
+  if (_sectionSize == 0)
+    return;
+
+  std::stringstream ss;
+  ss.write((char*) _pDataSection, _sectionSize);
```

Comment:
Why the cast?   Should _pDataSection be const char*?

Reply:
Cast isn't needed.  It has been removed.